tv_apps/fling-player/js/connector.js

```diff
@@ -1,125 +1,183 @@
+/* global evt, castingMessage, mDBG */
 (function(exports) {
   'use strict';
 
-  var seq = 0;
-  function Connector(player) {
-    this._player = player;
+  var noop = function () {};
+
+  /**
+   * This class handles the connection to controller via the presentation API
+   * The events of controller's message are:
+   *   - loadRequest: Triggered when controller asks to load video.
+   *           Will be passed in one object with url property storing the
+   *           video url.
+   *   - playRequest: Triggered when controller asks to play video.
+   *   - pauseRequest: Triggered when controller asks to pause video.
+   *   - seekRequest: Triggered when controller asks to seek video.
+   *           Will be passed in one object with time property storing the
+   *           seeking time.
+   *
+   * @param {object} presentation The navigator's presentation object
+   * @constructor
+   */
+  function Connector(presentation) {
+    this._presentation = presentation;
+    this._msgSeq = 0; // This sequence of message sent to controller
+    this._lastSeq = -1; // The sequence of the last message received
   }
 
-  var presentation = navigator.mozPresentation;
-  var proto = Connector.prototype;
+  var proto = evt(Connector.prototype);
 
-  proto.init = function c_init() {
-    if (!presentation) {
-      return;
-    }
+  proto.init = function () {
 
-    if (presentation.session) {
-      this.initSession(presentation.session);
-    } else {
-      presentation.addEventListener('sessionready', this);
+    if (this.init === noop) return;
+
+    mDBG.log('Connector#init');
+    if (!this._presentation) {
+      throw new Error('Init connection without the presentation object.');
     }
+
+    this.init = noop;
```

Comment:
1. Is there any situation that `init` would be called twice? Can we prevent that?
2. If running `init` twice unavoidable, can we use `this._initialized = true` rather than overriding `init`?